src/EditorFeatures/Core/Shared/Utilities/ThreadingContext.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Utilities;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
+{
+    [Export(typeof(IThreadingContext))]
+    [Shared]
+    internal sealed partial class ThreadingContext : IThreadingContext
```

Comment:
these classes need docs galore.  can you write that and then ping.  i can review then :)

Reply:
⌛️ Fixed locally and will be in next commit.